### PR TITLE
Fix, App Factory pattern

### PR DESCRIPTION
**Pull request by dpgaspar**
These changes include a new cli for FAB and a proper app factory pattern:

- New cli integrated with flask cli, so it's now easy to support app factory pattern and use the command line. All commands were "migrated":

```
export FLASK_APP="app:create_app('config')"
flask fab create-admin
flask fab list-users
....
flask run
```

- New, ``FAB_UPDATE_PERMS`` config key to enable or disable the automatic creation of permissions. Default is ``True``
- New, ``FAB_SECURITY_MANAGER_CLASS`` config key to set an alternative security manager class. Use it like an addon:

``` python
FAB_SECURITY_MANAGER_CLASS="app.sec.MySecurityManager"
```

- New cli command to create-permissions (can be done after app start on factory method):

```
flask fab create-permissions
```

Note, an example of init create_app (factory):

``` python
from flask import Flask
from flask_appbuilder import AppBuilder, SQLA

db = SQLA()
appbuilder = AppBuilder()


def create_app(config):
    app = Flask(__name__)
    with app.app_context():
        app.config.from_object(config)
        db.init_app(app)
        appbuilder.init_app(app, db.session)
        from . import views  # noqa
        appbuilder.post_init()
    return app
```

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -201,6 +201,15 @@ Use config.py to configure the following parameters. By default it will use SQLL
...
 +-----------------------------------+--------------------------------------------+-----------+
+| FAB_UPDATE_PERMS                  | Enables or disables update permissions     |           |
+|                                   | Default is True (Boolean)                  |   No      |
++-----------------------------------+--------------------------------------------+-----------+
+| FAB_ADD_SECURITY_VIEWS            | Enables or disables registering all        |           |
+|                                   | security views (boolean default:True)      |   No      |
++-----------------------------------+--------------------------------------------+-----------+
+| FAB_ADD_OPENAPI_VIEWS             | Enables or disables registering all        |           |
+|                                   | OPENAPI views (boolean default:True)       |   No      |
++-----------------------------------+--------------------------------------------+-----------+
```
Comment (craig-rueda) [inline docs/config.rst]: Add `FAB_SECURITY_MANAGER_CLASS` key

File: flask_appbuilder/api/__init__.py [hunk 2]
```diff
@@ -390,6 +390,9 @@ def _register_urls(self):
             attr = getattr(self, attr_name)
             if hasattr(attr, "_urls"):
                 for url, methods in attr._urls:
+                    log.debug(
+                        f"Registering route {self.blueprint.url_prefix}{url} {methods}"
```
Comment (craig-rueda) [inline flask_appbuilder/api/__init__.py]: 👍

File: flask_appbuilder/baseviews.py [hunk 3]
```diff
@@ -164,6 +164,9 @@ def _register_urls(self):
             attr = getattr(self, attr_name)
             if hasattr(attr, "_urls"):
                 for url, methods in attr._urls:
+                    log.debug(
+                        f"Registering route {self.blueprint.url_prefix}{url} {methods}"
```
Comment (craig-rueda) [inline flask_appbuilder/baseviews.py]: F Strings!!

File: flask_appbuilder/security/manager.py [hunk 4]
```diff
@@ -261,11 +263,27 @@ def __init__(self, appbuilder):
                 self.oauth_remotes[provider_name] = obj_provider
 
         # Setup Flask-Login
+        self.set_login_manager(app)
+
+        # Setup Flask-Jwt-Extended
+        self.set_jwt_manager(app)
+
+    def set_login_manager(self, app):
```
Comment (craig-rueda) [inline flask_appbuilder/security/manager.py]: Alternate naming / strategy:
```
def build_login_manager(self, app): -> LoginManager
```
This allows the `lm` to remain "private" and still be fully customizable

File: flask_appbuilder/security/manager.py [hunk 5]
```diff
@@ -261,11 +263,27 @@ def __init__(self, appbuilder):
...
+            Override to implement your custom login manager instance
+
+            :param app: Flask app
+        """
         self.lm = LoginManager(app)
         self.lm.login_view = "login"
         self.lm.user_loader(self.load_user)
 
-        # Setup Flask-Jwt-Extended
+    def set_jwt_manager(self, app):
```
Comment (craig-rueda) [inline flask_appbuilder/security/manager.py]: Alternate naming / strategy:
```
def build_jwt_manager(self, app): -> JWTManager
```
This allows the `jwt_manager` to remain "private" and still be fully customizable

File: flask_appbuilder/baseviews.py (hunk 3 above)
Comment (dpgaspar) [reply flask_appbuilder/baseviews.py]: FTW!! :)

File: docs/config.rst (hunk 1 above)
Comment (dpgaspar) [reply docs/config.rst]: Right! thks!

**Comment by mistercrunch** [comment]
304 files changed! Looks like you ran `black` on the code base. Makes for the changes here hard to review. Let's make sure the release notes have the breaking changes listed.

File: flask_appbuilder/baseviews.py (hunk 3 above)
Comment (craig-rueda) [reply flask_appbuilder/baseviews.py]: Make this `info`

**Comment by dpgaspar** [comment]
@mistercrunch changed it so that we don't have to break anything:

- `fabmanager` command will issue a deprecation warning, but can still be used.
- AppBuilder still accepts the named parameter `update_perms`

**Pull request merged**